python_modules/libraries/dagster-dbt/dagster_dbt/asset_utils.py

```diff
@@ -876,6 +876,24 @@ def build_dbt_multi_asset_args(
                     )
                 )
 
+            if dagster_dbt_translator.settings.enable_source_asset_checks:
```

Comment:
I think that this is the right logic for defining the check specs on the sources, I think it just needs to live in an independent function where we provide the set of asset keys which we want check specs for (since the asset keys here aren't actually part of the dbt multi asset).